kernel/kernel-api/src/main/java/io/delta/kernel/expressions/predicates/Predicate.java

```diff
@@ -0,0 +1,185 @@
+ *  </li>
+ *  <li>Name: <code>&gt;</code>
+ *   <ul>
+ *    <li>SQL semantic: <code>expr1 &gt; expr2</code></li>
+ *    <li>Since version: 3.0.0</li>
+ *   </ul>
+ *  </li>
+ *  <li>Name: <code>&gt;=</code>
+ *   <ul>
+ *    <li>SQL semantic: <code>expr1 &gt;= expr2</code></li>
+ *    <li>Since version: 3.0.0</li>
+ *   </ul>
+ *  </li>
+ *  <li>Name: <code>ALWAYS_TRUE</code>
+ *   <ul>
+ *    <li>SQL semantic: <code>Constant expression whose value is `true`</code></li>
+ *    <li>Since version: 3.0.0</li>
+ *   </ul>
+ *  </li>
+ *  <li>Name: <code>ALWAYS_FALSE</code>
+ *   <ul>
+ *    <li>SQL semantic: <code>Constant expression whose value is `false`</code></li>
+ *    <li>Since version: 3.0.0</li>
+ *   </ul>
+ *  </li>
+ *  <li>Name: <code>AND</code>
+ *   <ul>
+ *    <li>SQL semantic: <code>expr1 AND expr2</code></li>
+ *    <li>Since version: 3.0.0</li>
+ *   </ul>
+ *  </li>
+ *  <li>Name: <code>OR</code>
+ *   <ul>
+ *    <li>SQL semantic: <code>expr1 OR expr2</code></li>
+ *    <li>Since version: 3.0.0</li>
+ *   </ul>
+ *  </li>
+ * </ol>
+ */
+public class Predicate implements Expression {
+    private final String name;
+    private final List<Expression> children;
+
+    public Predicate(String name, List<Expression> children) {
+        final String nameUpper = name.toUpperCase(Locale.ENGLISH);
+        validatePredicateExpression(nameUpper, children);
+        this.name = nameUpper;
+        this.children = Collections.unmodifiableList(new ArrayList<>(children));
+    }
+
+    public String name() {
+        return name;
+    }
+
+    @Override
+    public List<Expression> children() {
+        return children;
+    }
+
+    @Override
+    public String toString() {
+        if (COMPARATOR_OPERATORS.contains(name)) {
+            return String.format("(%s %s %s)", children.get(0), name, children.get(1));
+        }
+        String args =
+            children.stream().map(Object::toString).collect(Collectors.joining(","));
+        return String.format("%s(%s)", name, args);
+    }
+
+    private static final Set<String> COMPARATOR_OPERATORS =
+        Stream.of("=", ">", ">=", "<", "<=").collect(Collectors.toSet());
+    /**
+     * List of supported predicate expressions
+     */
+    private static final Set<String> PREDICATE_TYPES = Stream.of(
+        "=",
+        "<",
+        "<=",
+        ">",
+        ">=",
+        "AND",
+        "OR",
+        "ALWAYS_TRUE",
+        "ALWAYS_FALSE"
+    ).collect(Collectors.toSet());
+
+    /**
+     * Validate the predicate name is in the allowed list and expected children types.
+     *
+     * @param name Predicate expression name
+     * @param children List of input expressions to the predicate expression.
+     */
+    private static void validatePredicateExpression(String name, List<Expression> children) {
+        checkArgument(PREDICATE_TYPES.contains(name), "Unknown predicate expression: %s", name);
+        switch (name) {
+            case "AND": // fall through
```

Comment:
What about a user provided expression like col1 AND col2? Do we expect the connector to create a different kind of expression than Predicate?

Reply:
`col1 AND col2` can be rewritten using `col1 = true AND col2 = true`. If this becomes a problem, we can relax this requirement in future.